### PR TITLE
[SPARK-23158] [SQL] Move HadoopFsRelationTest test suites to from sql/hive to sql/core

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
The test suites that extend HadoopFsRelationTest are not in sql/hive packages, but their directories are in sql/hive. We should move them to sql/core.

## How was this patch tested?
The existing tests.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala [hunk 1]
```diff
@@ -82,44 +80,4 @@ class OrcHadoopFsRelationSuite extends HadoopFsRelationTest {
...
-      spark.range(0, 10).write
-        .orc(file.getCanonicalPath)
-      val expectedCompressionKind =
-        OrcFileOperator.getFileReader(file.getCanonicalPath).get.getCompression
-      assert("SNAPPY" === expectedCompressionKind.name())
-    }
-  }
-}
-
-class HiveOrcHadoopFsRelationSuite extends OrcHadoopFsRelationSuite {
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala]: This is Hive only.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala [hunk 2]
```diff
@@ -82,44 +80,4 @@ class OrcHadoopFsRelationSuite extends HadoopFsRelationTest {
...
-      checkAnswer(df, copyDf)
-    }
-  }
-
-  test("Default compression codec is snappy for ORC compression") {
-    withTempPath { file =>
-      spark.range(0, 10).write
-        .orc(file.getCanonicalPath)
-      val expectedCompressionKind =
-        OrcFileOperator.getFileReader(file.getCanonicalPath).get.getCompression
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala]: `OrcFileOperator` is defined in `sql\hive`.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala [hunk 3]
```diff
@@ -82,44 +80,4 @@ class OrcHadoopFsRelationSuite extends HadoopFsRelationTest {
...
-        .orc(path)
-
-      // Check if this is compressed as ZLIB.
-      val maybeOrcFile = new File(path).listFiles().find { f =>
-        !f.getName.startsWith("_") && f.getName.endsWith(".zlib.orc")
-      }
-      assert(maybeOrcFile.isDefined)
-      val orcFilePath = maybeOrcFile.get.toPath.toString
-      val expectedCompressionKind =
-        OrcFileOperator.getFileReader(orcFilePath).get.getCompression
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala]: The same here.

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
cc @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala]: Thank you!

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcHadoopFsRelationSuite.scala]: @gatorsmile . This test case should be tested on `native` implementation, too.
`HiveOrcHadoopFsRelationSuite` test coverage is only `hive` implementation.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please keep the test coverage for the following two test cases for `native` ORCFileFormat.
```
test("SPARK-13543: Support for specifying compression codec for ORC via option()")
test("Default compression codec is snappy for ORC compression")
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonHadoopFsRelationSuite.scala [hunk 4]
```diff
@@ -110,14 +113,16 @@ class JsonHadoopFsRelationSuite extends HadoopFsRelationTest {
...
     import testImplicits._
-    withTempDir { tempDir =>
-      val path = tempDir.getAbsolutePath
-      Seq(badJson, """{"a":1}""").toDS().write.mode("overwrite").text(path)
-      val expected = s"""$badJson\n{"a":1}\n"""
-      val schema = new StructType().add("a", IntegerType).add("_corrupt_record", StringType)
-      val df =
-        spark.read.format(dataSourceName).option("multiLine", true).schema(schema).load(path)
-      checkAnswer(df, Row(null, expected))
+    withSQLConf(SQLConf.MAX_RECORDS_PER_FILE.key -> "2") {
```
Comment (jiangxb1987) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonHadoopFsRelationSuite.scala]: Just curious, why this change?

Comment (gatorsmile) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonHadoopFsRelationSuite.scala]: The test will fail if `SQLConf.MAX_RECORDS_PER_FILE.key` is less than 2

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonHadoopFsRelationSuite.scala]: I think the default value won't be less than 2, we don't need to be so careful...

**Comment by dongjoon-hyun** [comment]
Please don't format [this comment](https://github.com/apache/spark/pull/20331#pullrequestreview-90358792) during next update, @gatorsmile . Thanks.
